### PR TITLE
[WIP] Update all dependencies test run

### PR DESCRIPTION
**Pull request by jdavcs**
This is a test run.

### CODE REVIEW

**Comment by ic4f** [comment]
This is incorrect. Rerunning...

File: lib/galaxy/dependencies/pinned-requirements.txt [hunk 1]
```diff
@@ -140,55 +146,55 @@ python-neutronclient==7.2.1; python_version >= "3.6"
...
 sortedcontainers==2.3.0
 sqlalchemy-migrate==0.13.0
 sqlalchemy-utils==0.36.7
-sqlalchemy==1.3.22; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0")
+sqlalchemy==1.3.23; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0")
 sqlitedict==1.7.0
 sqlparse==0.4.1; python_version >= "3.5"
-starlette==0.13.6; python_version >= "3.6" and python_version < "4.0"
-starlette-context==0.3.1; python_version >= "3.7"
+starlette-context==0.1.3
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pinned-requirements.txt:179]: @mvdbeek Do we need to add a `>=` pin for starlette-context too or is 0.1.3 fine?

File: lib/galaxy/dependencies/pinned-requirements.txt [hunk 2]
```diff
@@ -140,55 +146,55 @@ python-neutronclient==7.2.1; python_version >= "3.6"
...
 python-swiftclient==3.10.1
 python3-openid==3.2.0; python_version >= "3.0"
-pytz==2020.5; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version < "4" and python_version >= "3.6"
+pytz==2021.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version < "4" and python_version >= "3.6"
 pyuwsgi==2.0.19.1.post0
 pyyaml==5.4.1; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.6.0")
 rdflib-jsonld==0.5.0; python_version >= "3.6" and python_version < "4"
 rdflib==4.2.2; python_version >= "3.6" and python_version < "4"
-refgenconf==0.9.3
+refgenconf==0.10.0
```
Comment (jdavcs) [inline lib/galaxy/dependencies/pinned-requirements.txt:154]: Needs to be rerun against latest dev with refgenconf pinned.

File: lib/galaxy/dependencies/pinned-requirements.txt (hunk 1 above)
Comment (jdavcs) [reply lib/galaxy/dependencies/pinned-requirements.txt:179]: There've been many updates between these versions: https://starlette-context.readthedocs.io/en/latest/changelog.html 
Seems risky to use 0.1.3. I'll pin it to `>=0.3.1`

Comment (jdavcs) [reply lib/galaxy/dependencies/pinned-requirements.txt:179]: https://github.com/galaxyproject/galaxy/pull/11643

File: pyproject.toml [hunk 3]
```diff
@@ -33,7 +33,7 @@ cwltool = "==3.0.20201109103151"
 dictobj = "*"
 docutils = "*"
 Fabric3 = "*"
-fastapi = "*"
+fastapi = ">=0.63.0"
```
Comment (nsoranzo) [inline pyproject.toml:36]: I have investigated why poetry was choosing fastapi 0.1.17:
- 0.1.17 was the last version not pinning starlette, see https://github.com/tiangolo/fastapi/commit/3d2c0993c128d406f7f53c1ddbfc9b5d42c08275#
- the latest fastapi (0.63.0) [pins starlette ==0.13.6](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L35)
- the latest starlette version is 0.14.2

So I think that poetry chooses starlette 0.14.2 first, and then search for the latest compatible fastapi, which is 0.1.17. So :+1: to this pinning.

Comment (jdavcs) [reply pyproject.toml:36]: Yes! Verified:
```
   1: selecting starlette (0.14.2)
   1: fact: fastapi (0.63.0) depends on starlette (0.13.6) ...
   1: derived: not fastapi (==0.63.0)
   1: fact: fastapi (0.62.0) depends on starlette (0.13.6) ...
   1: derived: not fastapi (==0.62.0)
   ...
   1: fact: fastapi (0.1.17) depends on starlette (>=0.9.7) ...   
   1: selecting fastapi (0.1.17)
```
This is not a good approach. I think I would expect it to backtrack and try to downgrade previous nodes, 1 downgrading step at a time. So yes, "-vvv" is very useful! On the bright side, pinning solved it.

Comment (nsoranzo) [reply pyproject.toml:36]: > This is not a good approach.

The only possible optimal solutions are (starlette 0.13.6 and fastapi 0.63.0) or (starlette 0.14.2 and fastapi 0.1.17), and poetry has no way to know which one is "correct" (i.e. which package to prioritise), given that we depended on both without specifying a minimum version.

Comment (jdavcs) [reply pyproject.toml:36]: I think the number of downgrade steps is potentially a good heuristic: downgrade starlette by one minor version or fastapi by 62 minor versions. That said, I've never designed a SAT-solver, so there may be a reason why this heuristic was not used.

**Pull request closed without merging**